bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -140,6 +190,24 @@ public void propertyChange(PropertyChangeEvent evt) {
 						mm.addMarker(false, this, Severity.Warning, String.format("Observation reaches airmass %1.2f.", maxAirmass), v, a);
 					}
 				}
+
+				if (maxAirmass < airmassLimitMax && minAirmass > airmassLimitMin && !elevationInterval.isEmpty()) {
```

Comment:
You can use `.foreach()` as in `elevationInterval.foreach(i -> {if (maxAirmass ...) ...} )` but i dunno if it is much better really.  I think this is fine.